src/main/java/net/snowflake/client/core/SessionUtil.java

```diff
@@ -259,12 +265,19 @@ static SFLoginOutput openSession(
       }
     }
 
-    boolean isClientStoreTemporaryCredential =
-        asBoolean(loginInput.getSessionParameters().get(CLIENT_STORE_TEMPORARY_CREDENTIAL));
-    if (isClientStoreTemporaryCredential) {
-      CredentialManager.getInstance().fillCachedIdToken(loginInput);
+    if (authenticator.equals(ClientAuthnDTO.AuthenticatorType.USERNAME_PASSWORD_MFA)) {
+      if (Constants.getOS() == Constants.OS.MAC || Constants.getOS() == Constants.OS.WINDOWS) {
+        loginInput.getSessionParameters().put(CLIENT_REQUEST_MFA_TOKEN, true);
+      } else {
+        if (!loginInput.getSessionParameters().containsKey(CLIENT_REQUEST_MFA_TOKEN)) {
```

Comment:
what is the use of this piece of code? It seems like if Authenticator is USERNAME_PASSWORD_MFA and CLIENT_REQUEST_MFA_TOKEN is not set, you just explicitly set it to false.

Reply:
Oh yeah, actually we can remove the line, I added it before only for my manually debugging, like I can check the field.